diff --git a/tests/Makefile b/tests/Makefile
index 778c7d6..ed99cac 100644
--- a/tests/Makefile
+++ b/tests/Makefile
@@ -35,10 +35,11 @@ PYTHON ?= python3
 TESTARTEFACT := versionsTest
 
 DEBUGFLAGS += -g -Wno-c++-compat
-CPPFLAGS   += -I$(ZSTDDIR) -I$(ZSTDDIR)/common -I$(ZSTDDIR)/compress \
+CPPFLAGS_EXTRA += -I$(ZSTDDIR) -I$(ZSTDDIR)/common -I$(ZSTDDIR)/compress \
               -I$(ZSTDDIR)/dictBuilder -I$(ZSTDDIR)/deprecated -I$(PRGDIR) \
               -DZSTD_WINDOW_OVERFLOW_CORRECT_FREQUENTLY=1
 
+override CPPFLAGS += $(CPPFLAGS_EXTRA)
 ZSTDCOMMON_FILES := $(sort $(ZSTD_COMMON_FILES))
 ZSTDCOMP_FILES   := $(sort $(ZSTD_COMPRESS_FILES))
 ZSTDDECOMP_FILES := $(sort $(ZSTD_DECOMPRESS_FILES))
@@ -207,7 +208,7 @@ paramgrill : $(ZSTD_FILES) $(PRGDIR)/util.c $(PRGDIR)/timefn.c $(PRGDIR)/benchfn
 
 CLEAN += datagen
 datagen : $(PRGDIR)/datagen.c datagencli.c
-	$(LINK.c) $^ -o $@$(EXT)
+	$(LINK.c) $(CPPFLAGS) $^ -o $@$(EXT)
 
 CLEAN += roundTripCrash
 roundTripCrash: CFLAGS += $(MULTITHREAD)
@@ -263,7 +264,7 @@ clean:
 # valgrind tests validated only for some posix platforms
 #----------------------------------------------------------------------------------
 UNAME := $(shell uname)
-ifneq (,$(filter $(UNAME),Linux Darwin GNU/kFreeBSD GNU OpenBSD FreeBSD NetBSD DragonFly SunOS AIX))
+ifneq (,$(filter $(UNAME),Linux Darwin GNU/kFreeBSD GNU OpenBSD FreeBSD NetBSD DragonFly SunOS AIX OS/390))
 HOST_OS = POSIX
 
 .PHONY: test-valgrind
@@ -307,6 +308,7 @@ shortest: ZSTDRTTEST=  # remove long tests
 shortest: test-zstd
 
 .PHONY: check
+check : CPPFLAGS += $(CPPFLAGS)
 check: shortest
 
 .PHONY: fuzztest
