diff --git a/tests/Makefile b/tests/Makefile
index 778c7d6..d65014f 100644
--- a/tests/Makefile
+++ b/tests/Makefile
@@ -35,9 +35,10 @@ PYTHON ?= python3
 TESTARTEFACT := versionsTest
 
 DEBUGFLAGS += -g -Wno-c++-compat
-CPPFLAGS   += -I$(ZSTDDIR) -I$(ZSTDDIR)/common -I$(ZSTDDIR)/compress \
+CPPFLAGS_EXTRA += -I$(ZSTDDIR) -I$(ZSTDDIR)/common -I$(ZSTDDIR)/compress \
               -I$(ZSTDDIR)/dictBuilder -I$(ZSTDDIR)/deprecated -I$(PRGDIR) \
               -DZSTD_WINDOW_OVERFLOW_CORRECT_FREQUENTLY=1
+override CPPFLAGS += $(CPPFLAGS_EXTRA)
 
 ZSTDCOMMON_FILES := $(sort $(ZSTD_COMMON_FILES))
 ZSTDCOMP_FILES   := $(sort $(ZSTD_COMPRESS_FILES))
@@ -58,15 +59,20 @@ ZSTDMT_OBJ3 := $(subst $(ZSTDDIR)/decompress/,zstdmt_d_,$(ZSTDMT_OBJ2))
 ZSTDMT_OBJ4 := $(ZSTDMT_OBJ3:.c=.o)
 ZSTDMT_OBJECTS := $(ZSTDMT_OBJ4:.S=.o)
 
+UNAME := $(shell uname)
 # Define *.exe as extension for Windows systems
 ifneq (,$(filter Windows%,$(OS)))
 EXT =.exe
 MULTITHREAD_CPP = -DZSTD_MULTITHREAD
+ifneq (,$(filter $(UNAME), OS/390))
+MULTITHREAD_LD  = -pthread
+else
 MULTITHREAD_LD  =
+#endif
 else
 EXT =
 MULTITHREAD_CPP = -DZSTD_MULTITHREAD
-MULTITHREAD_LD  = -pthread
+MULTITHREAD_LD  = 
 endif
 MULTITHREAD = $(MULTITHREAD_CPP) $(MULTITHREAD_LD)
 
@@ -207,7 +213,7 @@ paramgrill : $(ZSTD_FILES) $(PRGDIR)/util.c $(PRGDIR)/timefn.c $(PRGDIR)/benchfn
 
 CLEAN += datagen
 datagen : $(PRGDIR)/datagen.c datagencli.c
-	$(LINK.c) $^ -o $@$(EXT)
+	$(LINK.c) $(CPPFLAGS) $^ -o $@$(EXT)
 
 CLEAN += roundTripCrash
 roundTripCrash: CFLAGS += $(MULTITHREAD)
@@ -263,7 +269,7 @@ clean:
 # valgrind tests validated only for some posix platforms
 #----------------------------------------------------------------------------------
 UNAME := $(shell uname)
-ifneq (,$(filter $(UNAME),Linux Darwin GNU/kFreeBSD GNU OpenBSD FreeBSD NetBSD DragonFly SunOS AIX))
+ifneq (,$(filter $(UNAME),Linux Darwin GNU/kFreeBSD GNU OpenBSD FreeBSD NetBSD DragonFly SunOS AIX OS/390))
 HOST_OS = POSIX
 
 .PHONY: test-valgrind
@@ -307,6 +313,7 @@ shortest: ZSTDRTTEST=  # remove long tests
 shortest: test-zstd
 
 .PHONY: check
+check : CPPFLAGS += $(CPPFLAGS)
 check: shortest
 
 .PHONY: fuzztest
