diff --git a/programs/benchzstd.c b/programs/benchzstd.c
index fa2659e..32dba6a 100644
--- a/programs/benchzstd.c
+++ b/programs/benchzstd.c
@@ -20,13 +20,13 @@
 /* *************************************
 *  Includes
 ***************************************/
-#include "platform.h"    /* Large Files support */
-#include "util.h"        /* UTIL_getFileSize, UTIL_sleep */
 #include <stdlib.h>      /* malloc, free */
 #include <string.h>      /* memset, strerror */
 #include <stdio.h>       /* fprintf, fopen */
 #include <errno.h>
 #include <assert.h>      /* assert */
+#include "platform.h"    /* Large Files support */
+#include "util.h"        /* UTIL_getFileSize, UTIL_sleep */
 
 #include "timefn.h"      /* UTIL_time_t */
 #include "benchfn.h"
diff --git a/programs/dibio.c b/programs/dibio.c
index d19f954..6f98b12 100644
--- a/programs/dibio.c
+++ b/programs/dibio.c
@@ -21,14 +21,14 @@
 /*-*************************************
 *  Includes
 ***************************************/
-#include "platform.h"       /* Large Files support */
-#include "util.h"           /* UTIL_getFileSize, UTIL_getTotalFileSize */
 #include <stdlib.h>         /* malloc, free */
 #include <string.h>         /* memset */
 #include <stdio.h>          /* fprintf, fopen, ftello64 */
 #include <errno.h>          /* errno */
 #include <assert.h>
 
+#include "platform.h"       /* Large Files support */
+#include "util.h"           /* UTIL_getFileSize, UTIL_getTotalFileSize */
 #include "timefn.h"         /* UTIL_time_t, UTIL_clockSpanMicro, UTIL_getTime */
 #include "../lib/common/mem.h"  /* read */
 #include "dibio.h"
diff --git a/programs/fileio.c b/programs/fileio.c
index 5338fa6..f94fab8 100644
--- a/programs/fileio.c
+++ b/programs/fileio.c
@@ -23,8 +23,6 @@
 /*-*************************************
 *  Includes
 ***************************************/
-#include "platform.h"   /* Large Files support, SET_BINARY_MODE */
-#include "util.h"       /* UTIL_getFileSize, UTIL_isRegularFile, UTIL_isSameFile */
 #include <stdio.h>      /* fprintf, open, fdopen, fread, _fileno, stdin, stdout */
 #include <stdlib.h>     /* malloc, free */
 #include <string.h>     /* strcmp, strlen */
@@ -33,6 +31,8 @@
 #include <errno.h>      /* errno */
 #include <limits.h>     /* INT_MAX */
 #include <signal.h>
+#include "platform.h"   /* Large Files support, SET_BINARY_MODE */
+#include "util.h"       /* UTIL_getFileSize, UTIL_isRegularFile, UTIL_isSameFile */
 #include "timefn.h"     /* UTIL_getTime, UTIL_clockSpanMicro */
 
 #if defined (_MSC_VER)
diff --git a/programs/util.c b/programs/util.c
index d69b72a..56bb622 100644
--- a/programs/util.c
+++ b/programs/util.c
@@ -16,12 +16,12 @@ extern "C" {
 /*-****************************************
 *  Dependencies
 ******************************************/
-#include "util.h"       /* note : ensure that platform.h is included first ! */
 #include <stdlib.h>     /* malloc, realloc, free */
 #include <stdio.h>      /* fprintf */
 #include <time.h>       /* clock_t, clock, CLOCKS_PER_SEC, nanosleep */
 #include <errno.h>
 #include <assert.h>
+#include "util.h"       /* note : ensure that platform.h is included first ! */
 
 #if defined(_WIN32)
 #  include <sys/utime.h>  /* utime */
diff --git a/programs/zstdcli.c b/programs/zstdcli.c
index bfe18c0..51eece3 100644
--- a/programs/zstdcli.c
+++ b/programs/zstdcli.c
@@ -27,14 +27,14 @@
 /*-************************************
 *  Dependencies
 **************************************/
-#include "platform.h" /* IS_CONSOLE, PLATFORM_POSIX_VERSION */
-#include "util.h"     /* UTIL_HAS_CREATEFILELIST, UTIL_createFileList */
 #include <stdlib.h>   /* getenv */
 #include <string.h>   /* strcmp, strlen */
 #include <stdio.h>    /* fprintf(), stdin, stdout, stderr */
 #include <errno.h>    /* errno */
 #include <assert.h>   /* assert */
 
+#include "platform.h" /* IS_CONSOLE, PLATFORM_POSIX_VERSION */
+#include "util.h"     /* UTIL_HAS_CREATEFILELIST, UTIL_createFileList */
 #include "fileio.h"   /* stdinmark, stdoutmark, ZSTD_EXTENSION */
 #ifndef ZSTD_NOBENCH
 #  include "benchzstd.h"  /* BMK_benchFiles */
